src/metabase/models/query_cache.clj

```diff
@@ -2,10 +2,16 @@
   "A model used to cache query results in the database."
   (:require
    [metabase.models.interface :as mi]
-   [toucan.models :as models]))
+   [methodical.core :as methodical]
+   [toucan.models :as models]
+   [toucan2.core :as t2]))
 
 (models/defmodel QueryCache :query_cache)
 
+(methodical/defmethod t2/primary-keys QueryCache
+  [_model]
+  [:query_hash])
```

Comment:
Can you actually use Toucan 2 with tables that have non-integer and non-`:id` primary keys? Yes! We never even bothered to define the PK here in the past because none of the Toucan things would have worked with it 💪